apps/api_web/lib/api_web/plugs/clear_metadata.ex

```diff
@@ -0,0 +1,26 @@
+defmodule ApiWeb.Plugs.ClearMetadata do
+  @moduledoc """
+  Clear Logger metadata at the start of a new request.
+
+  Bandit is more agressive about re-using processes than Cowboy was, which means we
+  can't rely on the old behavior of the Logger metadata being automatically cleared
+  when the process is terminated.
+
+  This takes a list of metadata keys to clear.
+  """
+  @behaviour Plug
+
+  @impl Plug
+  def init(keys_to_clear) do
+    for key <- keys_to_clear do
+      {key, nil}
+    end
+  end
+
+  @impl Plug
+  def call(conn, metadata) do
+    _ = Logger.metadata(metadata)
```

Comment:
Why not use [reset_metadata](https://hexdocs.pm/logger/Logger.html#reset_metadata/1)? Are we logging something we want to retain?

Reply:
Fair enough! My concern is that we'll add a key and forget to add it here, but if that becomes a problem, we can cross that bridge then.